hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestBackgroundContainerDataScanner.java

```diff
@@ -133,6 +136,16 @@ public void testUnhealthyContainersDetected() throws Exception {
     verifyContainerMarkedUnhealthy(deletedContainer, never());
   }
 
+  @Test
+  @Override
+  public void testUnhealthyContainersTriggersVolumeScan() throws Exception {
+    try (MockedStatic<StorageVolumeUtil> mockedStatic = mockStatic(StorageVolumeUtil.class)) {
```

Comment:
Instead of `mockStatic`, each container instance in this test is initialized with a mocked `HddsVolume` [object](https://github.com/apache/ozone/blob/9a445ed01eb8224ebccbc8c7a2712eba36651aa6/hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/ozoneimpl/TestContainerScannersAbstract.java#L80). We can assert the number of invocations of `HddsVolume#check` to verify it is scanned.

If this ends up being too much change we can keep the static mock, but its usually cleaner to avoid using that if possible.